radix-transaction-scenarios/src/scenarios/all_scenarios.rs

```diff
@@ -1,54 +1,422 @@
 use super::*;
+use radix_engine::blueprints::consensus_manager::*;
+use radix_engine::system::bootstrap::*;
+use radix_engine::system::system_db_reader::*;
+use radix_engine::updates::*;
+use radix_engine::vm::wasm::*;
+use radix_engine::vm::*;
+use radix_substate_store_interface::db_key_mapper::*;
+use radix_substate_store_interface::interface::*;
+use radix_transactions::errors::*;
+use radix_transactions::validation::*;
+use sbor::prelude::*;
 
-pub fn get_builder_for_every_scenario() -> AllScenariosIterator {
-    AllScenariosIterator::default()
-}
-
-#[derive(Default)]
-pub struct AllScenariosIterator {
-    index: usize,
-}
-
-impl Iterator for AllScenariosIterator {
-    type Item = Box<dyn FnOnce(ScenarioCore) -> Box<dyn ScenarioInstance>>;
-
-    fn next(&mut self) -> Option<Self::Item> {
-        self.index += 1;
-        match self.index {
-            1 => Some(Box::new(|core| {
-                transfer_xrd::TransferXrdScenarioCreator::create(core)
-            })),
-            2 => Some(Box::new(|core| {
-                radiswap::RadiswapScenarioCreator::create(core)
-            })),
-            3 => Some(Box::new(|core| metadata::MetadataScenario::create(core))),
-            4 => Some(Box::new(|core| {
-                fungible_resource::FungibleResourceScenarioCreator::create(core)
-            })),
-            5 => Some(Box::new(|core| {
-                non_fungible_resource::NonFungibleResourceScenarioCreator::create(core)
-            })),
-            6 => {
-                Some(Box::new(|core| {
-                    account_authorized_depositors::AccountAuthorizedDepositorsScenarioCreator::create(core)
-                }))
+use account_authorized_depositors::AccountAuthorizedDepositorsScenarioCreator;
+use account_locker::AccountLockerScenarioCreator;
+use fungible_resource::FungibleResourceScenarioCreator;
+use global_n_owned::GlobalNOwnedScenarioCreator;
+use kv_store_with_remote_type::KVStoreScenarioCreator;
+use max_transaction::MaxTransactionScenarioCreator;
+use metadata::MetadataScenario;
+use non_fungible_resource::NonFungibleResourceScenarioCreator;
+use non_fungible_resource_with_remote_type::NonFungibleResourceWithRemoteTypeScenarioCreator;
+use radiswap::RadiswapScenarioCreator;
+use royalties::RoyaltiesScenarioCreator;
+use transfer_xrd::TransferXrdScenarioCreator;
+
+#[derive(Clone, Copy, Debug, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub enum ScenarioRequirements {
+    /// These scenarios do not require any protocol updates and are expected to be run at genesis
+    /// and be compatible with it.
+    Genesis,
+    /// These scenarios require one or more protocol updates and this enum specifies the protocol
+    /// update that they require.
+    ProtocolUpdateUpTo(ProtocolUpdate),
```

Comment:
I'd kinda like us to just make genesis a protocol update, but I don't know how feasible that is.

Reply:
I think that it's certainly possible, but might be too big of change for this PR :D